src/library_fs.js

```diff
@@ -471,6 +471,11 @@ FS.staticInit();` +
     closeStream(fd) {
       FS.streams[fd] = null;
     },
+    dupStream(origStream, fd = -1) {
+      var stream = FS.createStream(origStream, fd);
+      stream.stream_ops?.dup?.(stream);
```

Comment:
Can `stream_ops` really be null here?

Reply:
I was surprised by this, but I believe there was a CI failure when I didn't include this, I can double check which one. Also, it's really nice to be able to use `?.`.

Reply:
Yeah, `test_unistd_dup_rawfs` fails here without it.